bytecode-verificator/bytecode_verificator.sh

```diff
@@ -0,0 +1,479 @@
+#!/usr/bin/env bash
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+if [[ "${TRACE-0}" == "1" ]]; then
+    set -o xtrace
+fi
+
+################################
+# Bytecode verification script #
+################################
+
+RED='\033[0;31m'
+ORANGE='\033[0;33m'
+GREEN='\033[0;32m'
+NC='\033[0m'
+
+zero_padding=$(printf '%0.1s' "0"{1..64})
+placeholder_padding=$(printf '%0.1s' "-"{1..64})
+
+# Prerequisite executables
+prerequisites=(jq yarn awk curl shasum uname bc)
+
+# Environment vailable required
+envs=(WEB3_INFURA_PROJECT_ID ETHERSCAN_TOKEN)
+
+# Commandline args required
+cmdargs=(solc_version remote_rpc_url contract config_json)
+
+# Vars
+constructor_calldata=""
+contract_config_name=""
+local_rpc_url=""
+
+function show_help() {
+    echo -e "$ORANGE Usage: $NC $0"\
+      "--solc-version 0.x.y" \
+      "--remote-rpc https://*" \
+      "--config-json ../deployed-*.json" \
+      "--contract Name" \
+      "[--constructor-calldata *]"\
+      "[--proxy]"\
+      "[--implementation]"
+}
+
+# Fork PID of Ganache
+fork_pid=0
+
+# Entry point
+main() {
+  check_prerequisites
+  check_envs
+
+  parse_cmd_args "$@"
+  check_compiler
+  start_fork
+
+  compile_contract
+  deploy_contract_on_fork
+  compare_bytecode
+}
+
+# Service functions
+
+function check_prerequisites() {
+  for p in "${prerequisites[@]}"
+  do
+    if ! [ -x "$(command -v "$p")" ]; then
+      _err "$p app is required but not found"
+    fi
+  done
+}
+
+function check_envs() {
+  set +u
+  for e in "${envs[@]}"
+  do
+    if [[ -z "${!e}" ]]; then
```

Comment:
`[[ -v "$e" ]]` ?

Reply:
not sure if I used it correctly, but it doesn't work for me

Reply:
`!` symbol is must here I believe

Reply:
It seems -v was introduced in bash 4+ while macos still uses version 3

Reply:
Yeah, since 4.2. 🍏🐛